packages/next/server/image-optimizer.ts

```diff
@@ -473,15 +477,35 @@ async function writeToCacheDir(
   await promises.writeFile(filename, buffer)
 }
 
+function getFileNameWithExtension(
+  url: string,
+  contentType: string | null
+): string | void {
+  const [urlWithoutQueryParams] = url.split('?')
```

Comment:
This should use `new URL(url). pathname` instead of manual string parsing

Reply:
sure ;)

Reply:
We already handle that here 

https://github.com/vercel/next.js/blob/97a140f7339d8dc99539f89d3a888de8f007390c/packages/next/server/image-optimizer.ts#L95

Reply:
ok, you mean to do this only for absolute URLs... cool I'll change that

Reply:
Hmm, now that I think about it, I guess we need to handle both external and internal URLs so maybe your solution is sufficient

Reply:
Maybe use `href` instead

Reply:
> Hmm, now that I think about it, I guess we need to handle both external and internal URLs so maybe your solution is sufficient

yeah, that why I did that ;) 

> Maybe use href instead

the href is actually equal to URL :)